x/interchainstaking/keeper/ibc_packet_handlers_test.go

```diff
@@ -771,6 +773,168 @@ func (suite *KeeperTestSuite) TestHandleWithdrawForUserLSM() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestHandleWithdrawRewards() {
+	val := addressutils.GenerateValAddressForTest().String()
+	user := addressutils.GenerateAddressForTestWithPrefix("quick")
+	tests := []struct {
+		name      string
+		setup     func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone)
+		msg       func(zone *icstypes.Zone) sdk.Msg
+		checks    func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone)
+		triggered bool
+		err       bool
+	}{
+		{
+			name:   "wrong msg",
+			setup:  func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawValidatorCommission{
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       true,
+		},
+		{
+			name: "wrong context",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.ConnectionId = ""
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
+			},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawDelegatorReward{
+					DelegatorAddress: user,
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       true,
+		},
+		// at this time, the waitgroup will be uint32_max so it's not gonna trigger redemption rate
```

Comment:
Can you explain this?

Reply:
I see in case `zone.WithdrawalWaitgroup = 0`, then if we continue minus it in this test case, the value of `zone.WithdrawalWaitgroup` will be equal to `uint32_max`, how about checking this in our logic? cc @joe-bowman

Reply:
Can you make a pr for this @phamminh0811